app/ide-desktop/lib/client/src/index.ts

```diff
@@ -78,28 +85,58 @@ class App {
              * should be used here instead. */
             electron.app.on('ready', () => {
                 logger.log('Electron application is ready.')
-                void this.main(windowSize)
+                void this.main(args, windowSize)
             })
             this.registerShortcuts()
         }
     }
 
+    /** Try to acquire the single-instance lock. If it fails, a new window will be opened in the
+     * original instance, so this instance should be closed. */
+    becomeSingleInstance() {
+        const didAcquireLock = electron.app.requestSingleInstanceLock(
+            fileAssociations.CLIENT_ARGUMENTS
+        )
+        if (didAcquireLock) {
+            electron.app.on('second-instance', (_event, _argv, _cwd, argv) => {
+                const args = config.CONFIG.clone()
+                // `chromeOptions` is ignored for now as it is unknown whether changes to
+                // command-line switches take effect after a window is already open.
```

Comment:
Does it mean we ignore chrome options at all?

Reply:
It's only ignored on new windows - `setChromeOptions` is called in `run()` - so it is called for the first window. i can maybe see whether setting different options does work for new windows, and if it doesn't then edit this comment to be more clear